### DISCUSSION TITLE
Installation Methode 3 Debian Bookworm Installing dependencies using pip gives error

### DISCUSSION THREAD

**Discussion by pilorain** [scs/smartmeter-datacollector#72] [Q&A]
As I did before when my Pi3 B+ was under Debian Bulleseye I would like to add smartmeter-datacollector to Home Assistant.
I forgot to check the Bullseye requirement before upgrading my Pi.
Now I have Debian Bookworm with Home Assistant installed and running successfully.
The command sudo apt install ./python3-smartmeter-datacollector_1.2.1-1_armhf.deb gives errors, see [Bookworm_install_errror.txt](https://github.com/user-attachments/files/17305143/Bookworm_install_errror.txt)
Does anyone have a workaround solution or shall I go back to Bullseye ?
Thanks for help and regards, Pierre

**Comment by raymar9** [comment]
You're right, the current deb packages are built for bullseye. However, your log shows some issues concerning the pip Python package install - unfortunately not the exact problem...

```
Installing dependencies using pip..dpkg: erreur de traitement du paquet python3-smartmeter-datacollector (--configure) :
 le sous-processus paquet python3-smartmeter-datacollector script post-installation installé a renvoyé un état de sortie d'erreur 1
```
I try to provide a new release soon with updated OS support.

In the meantime you might succeed when choosing the installation method directly working with `pip` and the Python package [smartmeter-datacollector](https://pypi.org/project/smartmeter-datacollector). However then, you then need to find a solution for the service starting upon boot-up (see [systemd service File](https://github.com/scs/smartmeter-datacollector/blob/master/smartmeter-datacollector.service) (change the username)) or write a DockerFile.

**Reply by pilorain** [reply]
Sorry for not investigate further on this _E: Sub-process /usr/bin/dpkg returned an error code (1)._

**Reply by raymar9** [reply]
Don't worry, I'll check that when creating a new release.

**Comment by pilorain** [comment]
The next day I started the whole installation again under Bullseye. That's when I discovered my big GPIO connection error see https://github.com/scs/smartmeter-datacollector/issues/60#issuecomment-2408642888.

Now everything works perfectly: the LG-E450 meter integrated into Home Assistant.

Many thanks for this open source software. Pierre
